File: 01-fibonacci/03.py

def fib(n):
    table = [0, 1]

    for i in range(2, n+1):
        table.append(table[i-1] + table[i-2])

    return table[n]


print(fib(4))
print(fib(5))
print(fib(6))
print(fib(50))  # Now we wont have stackoverflow, and also neednot pass the table

File: 01-fibonacci/03.cpp
#include <iostream>
#include <vector>

#define ll long long int

using namespace std;

ll fib(int n)
{
    vector<ll> table = {0, 1};

    for (int i = 2; i <= n; ++i)
        table.push_back(table[i - 1] + table[i - 2]);

    return table.back();
}

int main()
{

    cout << fib(6) << endl;
    cout << fib(7) << endl;
    cout << fib(8) << endl;
    cout << fib(50) << endl;

    return 0;
}